ecosystem/indexer-grpc/indexer-grpc-table-info/src/internal_indexer_db_service.rs

```diff
@@ -57,18 +58,42 @@ impl InternalIndexerDBService {
         Arc::clone(&self.db_indexer)
     }
 
-    pub async fn run(&mut self) {
-        let mut start_version = self
-            .db_indexer
-            .indexer_db
-            .get_persisted_version()
-            .unwrap_or(0);
+    pub async fn get_start_version(&self, node_config: &NodeConfig) -> Result<Version> {
+        let indexer_version = self.db_indexer.indexer_db.get_persisted_version()?;
+        let fast_sync_enabled = node_config
+            .state_sync
+            .state_sync_driver
+            .bootstrapping_mode
+            .is_fast_sync();
+        let mut db_min_version = self.db_indexer.get_main_db_lowest_viable_version()?;
+
+        // Wait till fast sync is done
+        while fast_sync_enabled && db_min_version == 0 {
+            tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+            db_min_version = self.db_indexer.get_main_db_lowest_viable_version()?;
+        }
+        let fast_sync_version_opt = self.db_indexer.indexer_db.get_fast_sync_version()?;
+        if indexer_version >= db_min_version {
```

Comment:
still need to detect the case where state indices being turned on after the fact (so without a base) ?
cc @grao1991

Reply:
For the without base case, the indexer version would be 0  and db_min_version would be > 0 and fast_sync_version_opt would be None?

Reply:
I mean state indices being turned on after the other indices has already been running. in that case indexer_version can be larger than db_min_version?